cypress/integration/addSource.spec.ts

```diff
@@ -0,0 +1,19 @@
+describe("Add source feature in LanguageSource page", function () {
+  it("should find a button to press to add source by uploading file", function () {
+    cy.visit("/languages");
+    cy.get("button").contains("Sources").click();
+
+    // Add source component should show after clicking the Add Source Button
+    cy.get("button").contains("Add Source").click().scrollIntoView();
+    cy.get("[data-cy=add-source-window]").should("exist");
+
+    cy.get("button").contains("UPLOAD").click();
```

Comment:
Since you are already using `data-cy` attribute throughout your test, I would suggest adding it to the button as well.

Reply:
For me, data-cy is only used for selecting div (contains more than one components). Some components created by us, such as <Button />, data-cy is an unknown attribute, so it might be easier to select using "button"?

Reply:
I agree; please add the `data-cy` attribute and use it to select in the Cypress test. See here: https://docs.cypress.io/guides/references/best-practices.html#Selecting-Elements

Reply:
Yup, you can add testing attribute to the reusable button component instead of just selecting selecting the button.

Reply:
Sure!